script.js

```diff
@@ -2,11 +2,98 @@
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  const mainEle = document.createElement('div');
+  mainEle.id = 'main'
+  rootElem.appendChild(mainEle)
+
+  //creating eposides and display them with their names and summary.
+  for (let i = 0 ; i < episodeList.length ; i++ ){
+    divEle = document.createElement('div');
+   divEle.id = 'episode'
+   mainEle.appendChild(divEle)
+   let titleEle = document.createElement('h1')
+   titleEle.id = 'title'
+   divEle.appendChild(titleEle)
+   titleEle.innerText = nameOfEpisodes(episodeList[i])
+   let EpisodeImage = document.createElement('img')
+      EpisodeImage.src = episodeList[i].image.medium;
+   divEle.appendChild(EpisodeImage);
+   let paraEle = document.createElement('div')
+   paraEle.id = 'para'
+   paraEle.innerHTML = episodeList[i].summary
+   divEle.appendChild(paraEle)
+
+
+  }
+
+
+  ////// create the name of the episodes
+  function nameOfEpisodes(film){
+    return`${film.name}-S${
+      film.season.toString().padStart(2, '0')}E${
+      film.number.toString().padStart(2, '0')}`
+  }
+  //////////////// displaying numbers of Episodes
+  function displayingNumOfEpisodes(num){
+    const displaying = document.getElementById('displaying')
+    displaying.innerText = `Displaying ${num.length}/${episodeList.length}`
 }
+displayingNumOfEpisodes(episodeList)
+  //////////filter movies
+  function filterMovies(){
+    return  show = arrayOfEpisodes.filter(f=> f.style.display === 'block')
+    }
+
+ ///////////// live seach 
+ let list = document.querySelectorAll('#episode')
+ let arrayOfEpisodes = Array.from(list)
+    
+     ///////////////////function of live search
+  const searchBar = document.forms['search'].querySelector('input');
+  searchBar.addEventListener('keyup', function(e){
+   const term = e.target.value.toLowerCase()
+   arrayOfEpisodes.forEach(film=> {
+         const movieText = film.innerHTML
+      if(movieText.toLowerCase().indexOf(term) != -1){
+        film.style.display = 'block'
+      } else {
+        film.style.display = 'none'
+      }
+    })
+    displayingNumOfEpisodes(filterMovies())
+ })
+
+     ///////// Selection bar
+     let selectionBar = document.getElementById('selection')
+     let option = document.createElement('option')
+     option.innerHTML = 'Select all movies'
+     selectionBar.appendChild(option)
+     ///////////////drop down list 
+     episodeList.forEach(movie=> {
+    const movieOption = document.createElement('option')
+    movieOption.innerText = nameOfEpisodes(movie)
+    selectionBar.appendChild(movieOption)
+     })
+     //////////function for select bar
+     selectionBar.addEventListener('change', function(e){
+     let select = e.target.value
+     arrayOfEpisodes.forEach(movie=> {
+     let selectHeading = movie.firstElementChild.innerText
+     if(select === selectHeading || select === 'Select all movies') {movie.style.display = 'block'}
+     else {movie.style.display = 'none'}
+     });
```

Comment:
you can also search by the episode id which is unique number to show the selected one.